EasyCommands/EasyCommands/Commands/CommandGroupDelegate.cs

```diff
@@ -51,18 +62,42 @@ public override void Invoke(TSender sender, string args)
         {
             if(args.Length == 0)
             {
-                throw new CommandParsingException($"{string.Format(Context.TextOptions.ShowSubcommands, Name)}\n{SubcommandList()}");
+                // Check for default command
+                if (defaultCommand != null)
+                {
+                    Context.CommandHandler.PreCheck(sender, this);
+                    defaultCommand.Invoke(sender, "");
+                }
+                else
+                {
+                    throw new CommandParsingException($"{string.Format(Context.TextOptions.ShowSubcommands, Name)}\n{SubcommandList()}");
+                }
             }
-            (string subcommand, string subcommandArgs) = args.SplitAfterFirstSpace();
-            if(!subcommands.ContainsKey(subcommand))
+            else
             {
-                throw new CommandParsingException(
-                    string.Format(Context.TextOptions.CommandNotFound, $"{Name} {subcommand}") + "\n"
-                    + string.Format(Context.TextOptions.ShowSubcommands, Name) + "\n"
-                    + SubcommandList());
+                (string subcommand, string subcommandArgs) = args.SplitAfterFirstSpace();
+                if(!subcommands.ContainsKey(subcommand))
+                {
+                    // Check for default command
+                    if (defaultCommand != null)
+                    {
+                        Context.CommandHandler.PreCheck(sender, this);
+                        defaultCommand.Invoke(sender, args);
+                    }
+                    else
+                    {
+                        throw new CommandParsingException(
+                            string.Format(Context.TextOptions.CommandNotFound, $"{Name} {subcommand}") + "\n"
+                            + string.Format(Context.TextOptions.ShowSubcommands, Name) + "\n"
+                            + SubcommandList());
+                    }
+                }
+                else
+                {
+                    Context.CommandHandler.PreCheck(sender, this);
+                    subcommands[subcommand].Invoke(sender, subcommandArgs);
+                }
             }
-            Context.CommandHandler.PreCheck(sender, this);
-            subcommands[subcommand].Invoke(sender, subcommandArgs);
         }
 
         public string SubcommandList()
```

Comment:
Can you make sure that any user messages (error messages, the HelpCommand in the Example project) generated by this system show reasonable looking messages for default subcommands? I believe you'd have to change syntaxDocumentation in BaseCommandDelegate.